tests/tracking/test_rest_tracking.py

```diff
@@ -240,16 +239,16 @@ def test_log_metrics_params_tags(mlflow_client):
     assert metric.key == "metric"
     assert metric.value == 123.456
     assert metric.timestamp == 789
+    assert metric.step == 2
 
 
 def test_log_batch(mlflow_client):
     experiment_id = mlflow_client.create_experiment('Batch em up')
     created_run = mlflow_client.create_run(experiment_id)
     run_id = created_run.info.run_uuid
-    # TODO(sid): pass and assert on step
     mlflow_client.log_batch(
```

Comment:
Test logging a metric with a non-zero step via the `MlflowClient.log_batch` API (this doesn't actually test new functionality added by this PR - the same test would have passed after we merged thee code changes in #1142)